pkg/engine2/path_selection/edge_validity.go

```diff
@@ -155,7 +161,9 @@ func checkIfPropertyContainsResource(property interface{}, resource construct.Re
 	return false
 }
 
-func checkIfLoneDependency(ctx solution_context.SolutionContext, resource, toCheck construct.ResourceId, direction knowledgebase.Direction) (bool, error) {
+func checkIfLoneDependency(ctx solution_context.SolutionContext,
+	resource, toCheck construct.ResourceId, direction knowledgebase.Direction,
+	selector knowledgebase.ResourceSelector) (bool, error) {
```

Comment:
```suggestion
func checkIfLoneDependency(
	ctx solution_context.SolutionContext,
	resource, toCheck construct.ResourceId,
	direction knowledgebase.Direction,
	selector knowledgebase.ResourceSelector
) (bool, error) {
```
nit: I prefer this style of multi-line definition, but we haven't standardized anything